python/cudf/cudf/core/dataframe.py

```diff
@@ -217,6 +217,25 @@ def __init__(self, data=None, index=None, columns=None, dtype=None):
                         ),
                     )
                 )
+        elif hasattr(data, "__array_interface__"):
+            arr_interface = data.__array_interface__
+            if len(arr_interface["descr"]) == 1:
+                # not record arrays
+                new_df = self._from_arrays(data, index=index, columns=columns)
+            else:
+                new_df = self.from_records(data, index=index, columns=columns)
+            self._data = new_df._data
+            self._index = new_df._index
+            self.columns = new_df.columns
+        elif hasattr(data, "__cuda_array_interface__"):
+            arr_interface = data.__cuda_array_interface__
+            if len(arr_interface["descr"]) == 1:
+                new_df = self._from_arrays(data, index=index, columns=columns)
+            else:
+                new_df = self.from_records(data, index=index, columns=columns)
+            self._data = new_df._data
+            self._index = new_df._index
+            self.columns = new_df.columns
```

Comment:
Move this above the `__array_interface__` block to short circuit that check in case of a GPU object.